### ISSUE TITLE
The steps of model training

### ISSUE THREAD

**Issue by 09876qwrte** [HsiehYiChia/canny_text#6]
Hi, Thank you for your work , Can you tell me The steps of model training more  in detail , which function will be used, Thank you

**Comment by 09876qwrte** [comment]
I only want to train the part of detection , without the OCR recognition

**Comment by HsiehYiChia** [comment]
You need to transform your data to LBP by get_lbp_data(). After that, feed the LBP data to train_cascade(). 
I'll fix this to make it easier in someday :P

**Comment by 09876qwrte** [comment]
So，there only use get_lbp_data() and train_cascade() in the training?   I have train the model by your data yesterday for 16 hours, but there have nothing result and log, Is this normal?

**Comment by HsiehYiChia** [comment]
Yes, for the training of detection classifier, it only involves
- **_get_lbp_data_**  (get lbp data of training data)
- **_train_cascade_**  (train weak and strong text classifiers)

I can't recall the detail of them right now, but both functions should finish within few minutes. 
The code isn't too long to trace, so you can trace it and see how it works

**Comment by 09876qwrte** [comment]
I found opencv_train() in your code , could this function take place of train_cascade(). and have the same  result?

**Comment by HsiehYiChia** [comment]
This function will train Adaboost classifier for OpenCV's build-in Machine learning module, and that won't be compatible with my Adaboost. I add this function just for comparison between my Adaboost and OpenCV's Adaboost.

**Comment by 09876qwrte** [comment]
Hi , I want to get the Binary MASK for every ER*,  Can this be implemented in your code?

**Comment by HsiehYiChia** [comment]
That is feasible, I had already try that. However, both time and space complexity will increase, especially space complexity because there are enormous amount of ER in an image.

**Comment by 09876qwrte** [comment]
I want to put the Binary MASK to CNN, so could you tell where you have realize it in your code,Thank you

**Comment by HsiehYiChia** [comment]
There are 2 way you can use:
1. Use a link list of pixels for each ER. Whenever a pixel is accumulate to this ER, append the pixel to the link list. It is implemented in **er_accumulate** or **er_tree_extract**. You can utilize **int pixel** and  **struct plist ** of **struct ER**.  A brief intro. of **int pixel** is it stands for the position of a pixel:   
_**pixel%image_width = x ; pixel/image_width = y**_     
2. Another work around is to binarize the ER since we keep tracking the level of every ER. You can utilize **int level** of **struct ER**. Keep in mind that this method may not give you exact pixel mask because the bounding box of ER could contain something which is not a text.

**Comment by 09876qwrte** [comment]
Thank you very much, And I want to continue training on the basis of the last training result, so I would like to ask if I can use the last training result to initialize the current training model, If this way is ok, what should to change

**Comment by HsiehYiChia** [comment]
I am sorry, I don't get it. What do you mean "training on the basis of the last training result"?

**Comment by 09876qwrte** [comment]
the last training result is the "strong.classifier" and "weak.classifier", The "training on the basis of the last training result" is like the finetune in the deep learning
